### ISSUE TITLE
ActiveSupport::MessageEncryptor fails in 5.0.2 with undefined method `call' for nil:NilClass Did you mean?  caller org/jruby/RubyMarshal.java:145:in `load'

### ISSUE THREAD

**Issue by rammpeter** [rails/rails#28876]
### Steps to reproduce
ActiveSupport::MessageEncryptor.decrypt_and_verify fails always in 5.0.2 with 
undefined method `call' for nil:NilClass
Did you mean?  caller
org/jruby/RubyMarshal.java:145:in `load'

also the example from http://api.rubyonrails.org/classes/ActiveSupport/MessageEncryptor.html fails this way:

```
salt  = SecureRandom.random_bytes(64)
key   = ActiveSupport::KeyGenerator.new('password').generate_key(salt, 32) 
crypt = ActiveSupport::MessageEncryptor.new(key)                           
encrypted_data = crypt.encrypt_and_sign('my secret data')                 
crypt.decrypt_and_verify(encrypted_data)    

```

### Expected behavior
return decrypted value instead of exception

### Actual behavior
Exception:
undefined method `call' for nil:NilClass
Did you mean?  caller
org/jruby/RubyMarshal.java:145:in `load'

### System configuration
**Rails version**:
5.0.2
**Ruby version**:
jruby 9.1.8.0

**Comment by pixeltrix** [comment]
@rammpeter are you using the jruby-openssl gem?

**Comment by rammpeter** [comment]
@pixeltrix  
Yes I've installed the gem jruby-openssl, but it is not named in Gemfile.lock of the rails project I did run "rails console" for.

**Comment by pixeltrix** [comment]
Is it in your Gemfile? If not it won't be included when you run `bundle exec` - since it isn't in your lock file it suggests that it isn't

**Comment by rammpeter** [comment]
jruby-openssl ist neither in Gemfile nor in Gemfile.lock.
Do you recommend to include it into Gemfile to make ActiveSupport::MessageEncryptor.decrypt_and_verify working?

**Comment by pixeltrix** [comment]
@rammpeter yes, try it with it in the Gemfile please.

**Comment by rammpeter** [comment]
There 's no change in behavior wether I include jruby-openssl in Gemfile / Gemfile.lock or not.

What do I do? :
- Mac-OS Sierra
- jruby-9.1.8.0 installed via rvm
- rails 5.0.2 gems loaded
- > rails new bla
- bla> rails console
- console> execute above example for MessageEncryptor

**Comment by pixeltrix** [comment]
@rammpeter looks like it's a problem with `Marshal` in Rails:

```
$ rails c
Loading development environment (Rails 5.0.2)
>> Marshal.load(Marshal.dump("my secret data"))
NoMethodError: undefined method `call' for nil:NilClass
```

Looks like it might be something we're doing since just using `jirb` works:

```
$ jirb
>> Marshal.load(Marshal.dump("my secret data"))
=> "my secret data"
```

A workaround for the moment is to use `JSON` as the serializer if that's okay, e.g:

```
$ rails c
>> salt = SecureRandom.random_bytes(64)
>> key = ActiveSupport::KeyGenerator.new('password').generate_key(salt, 32)
>> crypt = ActiveSupport::MessageEncryptor.new(key, serializer: JSON)
>> encrypted_data = crypt.encrypt_and_sign('my secret data')
>> crypt.decrypt_and_verify(encrypted_data)
=> "my secret data"
```

**Comment by pixeltrix** [comment]
Turns out this is a known bug that's fixed in the upcoming 9.1.9.0 release of jRuby - see https://github.com/jruby/jruby/issues/4526
